### ISSUE TITLE
about the pre-computed average image in pre-trained imagenet models

### ISSUE THREAD

**Issue by tinalegre** [vlfeat/matconvnet#371]
Hi guys,

I recently downloaded a imagenet pretrained model from the matconvnet models [repository](http://www.vlfeat.org/sandbox-matconvnet/models/), namely: `imagenet-matconvnet-vgg-m.mat` (alias: `model1`). Before the v17.0 release, I have been using: `imagenet-vgg-m.mat` (alias: `model2`). What I realized is that the `average image` from `model 1` (i.e. model1.meta.normalization.averageImage) is a vector of size 3x1 instead of a matrix of size 224x224x3 as in model2 (i.e. model2.meta.normalization.averageImage). It looks like a bug. Can anyone doublecheck this information? The same happened with verydeep-16 model, I haven't compared other models though, e.g. vgg-f, vgg-s. Any ideas are welcomed ;-)

**Comment by fengyunxiaozi** [comment]
it  seems the model "imagenet-matconvnet-vgg-m.mat" compute the mean value of each channel respectively, thus the result is 3*1 vector. but I'm not sure about this. I haven't seen any official documents about this yet.

**Comment by tinalegre** [comment]
@fengyunxiaozi i also thought, it was the mean value of each channel. I computed those values from the [224,224,3] image but they didnt match. In addition, even those values represent the per-channel average values, it should not be the case. what do you think @lenck ?

**Comment by wendygo** [comment]
I also meet the same problem.The size of "meta.normalization.averageImage" are 1*1*3 in imagenet-vgg-verydeep-16 and imagenet-matconvnet-vgg-verydeep-16 which provided in website...

**Comment by vedaldi** [comment]
Hi, this is correct. Many of the recent models use a 1 x 1 average image (i.e. just an average color). In this manner they can be applied convolutional to images of arbitrary size, which is sometimes useful (e.g. for segmentation).

Just use `bsxfun(@minus, image, meta.normalization.averageImag)` to subtract instead of `image - meta.normalization.averageImage`.

**Comment by fengyunxiaozi** [comment]
@tinalegre  yes,  they didnt match, but they are  close. I think this may because the training images are different when computing the  mean .  see the answer of vedaldi, it is average color. So that is average of the R channel, G channel and B channel, respectively! Right?

**Comment by vedaldi** [comment]
Hi, yes, it is just like a 1x1 pixels image…

**Comment by tinalegre** [comment]
@vedaldi thank you! intuitively one would then understand that the spatial distribution/configuration of colors is not relevant, right? it's but a bit strange to me. which is the reasoning behind it? especially as in the case of image segmentation, the spatial distribution plays a role, e.g. sky over grass on the upper-top of the image. do you might have a reference (cvpr/iccv) on models that use 1x1 avg images?
